python/cudf/cudf/core/column/column.py

```diff
@@ -119,25 +115,19 @@ def __repr__(self):
     def to_pandas(
         self, index: ColumnLike = None, nullable: bool = False, **kwargs
     ) -> "pd.Series":
-        if nullable and self.dtype in cudf_dtypes_to_pandas_dtypes:
-            pandas_nullable_dtype = cudf_dtypes_to_pandas_dtypes[self.dtype]
-            arrow_array = self.to_arrow()
-            pandas_array = pandas_nullable_dtype.__from_arrow__(arrow_array)
-            pd_series = pd.Series(pandas_array, copy=False)
-        elif str(self.dtype) in NUMERIC_TYPES and self.null_count == 0:
-            pd_series = pd.Series(cupy.asnumpy(self.values), copy=False)
-        elif is_interval_dtype(self.dtype):
-            pd_series = pd.Series(
-                pd.IntervalDtype().__from_arrow__(self.to_arrow())
-            )
-        else:
-            pd_series = self.to_arrow().to_pandas(**kwargs)
+        """Convert object to pandas type.
+
+        The default implementation falls back to PyArrow for the conversion.
+        """
+        pd_series = self.to_arrow().to_pandas(**kwargs)
```

Comment:
In the previous version of this function, most of the logic was only really applicable to numeric types (and is now [here](https://github.com/rapidsai/cudf/pull/8081/files#diff-7ffeb7660d604ebf1ce745d0c3727e2c9e4e05f9226536f17b46e81095f4432fR746)). That branch explicitly handled null values in a special way. The line that's still here was the default fallback for all types and assumes that `to_arrow` handles nulls appropriately (I assume `pyarrow.Table.to_pandas` does the right thing afterwards). If this generally makes sense as a fallback option for `to_pandas`, then I think we have to implicitly rely on our `to_arrow` methods handling null. Our base implementation of `to_arrow` calls through to a `libcudf` interop API that I assume handles nulls correctly. I don't think we have any way of enforcing this generically on subclasses that override `to_arrow`, though.

Reply:
Just chatted with @shwina. There isn't really a sensible default way to handle the nullable parameter. Since the only use of `ColumnBase.to_pandas` is in `cudf.Series.to_pandas`, we can control the use of this API internally and we won't have any problems here as long as classes that have reasonable corresponding null behavior in pandas handle this parameter correctly in the subclass (which I believe @shwina said is just integral types in pandas right now).

Reply:
I believe `cudf.Index.to_pandas` uses this as well.

Reply:
Yes, it does, but in that case it's even simpler because the values are the index so there's no additional index parameter.